src/coreclr/jit/valuenum.cpp

```diff
@@ -6001,6 +6001,84 @@ void ValueNumStore::SetVNIsCheckedBound(ValueNum vn)
     m_checkedBoundVNs.AddOrUpdate(vn, true);
 }
 
+#ifdef FEATURE_HW_INTRINSICS
+ValueNum ValueNumStore::EvalHWIntrinsicFunUnary(
+    var_types type, NamedIntrinsic ni, VNFunc func, ValueNum arg0VN, bool encodeResultType, ValueNum resultTypeVN)
+{
+    if (IsVNConstant(arg0VN))
+    {
+        switch (ni)
+        {
+#ifdef TARGET_ARM64
+            case NI_ArmBase_LeadingZeroCount:
+#else
+            case NI_LZCNT_LeadingZeroCount:
+#endif
+            {
+                UINT32 cns = (UINT32)GetConstantInt32(arg0VN);
+                int    lzc = 0;
+                while (cns != 0)
+                {
+                    cns = cns >> 1;
+                    lzc++;
+                }
+                return VNForIntCon(32 - lzc);
+            }
+
+#ifdef TARGET_ARM64
+            case NI_ArmBase_Arm64_LeadingZeroCount:
+#else
+            case NI_LZCNT_X64_LeadingZeroCount:
+#endif
+            {
+                UINT64 cns = (UINT64)GetConstantInt64(arg0VN);
+                int    lzc = 0;
+                while (cns != 0)
+                {
+                    cns = cns >> 1;
+                    lzc++;
+                }
+                return VNForIntCon(64 - lzc);
+            }
+
+            default:
+                break;
+        }
+    }
+
+    if (encodeResultType)
+    {
+        return VNForFunc(type, func, arg0VN, resultTypeVN);
+    }
+    return VNForFunc(type, func, arg0VN);
+}
+
+ValueNum ValueNumStore::EvalHWIntrinsicFunBinary(var_types      type,
+                                                 NamedIntrinsic ni,
+                                                 VNFunc         func,
+                                                 ValueNum       arg0VN,
+                                                 ValueNum       arg1VN,
+                                                 bool           encodeResultType,
+                                                 ValueNum       resultTypeVN)
+{
+    // E.g. for "Sse2.Or(cnsVec1, cnsVec2)" it looks like this:
+    // case NI_SSE2_Or:
+    //{
+    //    simd16_t v1 = GetConstantSimd16(arg0VN);
+    //    simd16_t v2 = GetConstantSimd16(arg1VN);
+    //    v1.i64[0] |= v2.i64[0];
+    //    v1.i64[1] |= v2.i64[1];
+    //    return VNForSimd16Con(v1);
+    //}
```

Comment:
Is this comment missing some context? Seems out of place

Reply:
I left an example of a folding for a binary HWIntrinsic VNFunc for someone who decides to implement it here. (E.g. LLVM folds all vector ops)

Reply:
Do you plan to add it? Otherwise I don't really see a reason to create this empty skeleton which might just stay like this forever.

Reply:
No near future plans although I did see hits in libraries.pmi for Intrinsics with two VNConstants.
Ok I'll remove it